### PR TITLE
Change to noarch python build

### PR DESCRIPTION
**Pull request by jayqi**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,20 +11,17 @@ source:
...
+  noarch: python
   script: {{ PYTHON }} -m pip install . -vv
 
 requirements:
-  build:
-    - python                                 # [build_platform != target_platform]
-    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
   host:
     - pip
     - python
```
Comment (bollwyvl) [inline recipe/meta.yaml]: need python pins in `host` as well

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,20 +11,17 @@ source:
   sha256: 7135f975b28917348f3ca8feb85f83277c89ac9acf331d1b87d945ecdbe2dffb
 
 build:
-  number: 0
-  skip: true  # [py<36]
+  number: 1
+  noarch: python
```
Comment (bollwyvl) [inline recipe/meta.yaml:15]: _every_ output also needs `noarch: python` as well. you _may_ prefer use yaml "anchors" to reduce some of the drudgery, e.g.

```yaml
build: &build
  number: 1
  noarch: python

outputs:
  - name: foo
    build: *build
```

**Comment by jayqi** [comment]
@bollwyvl Thanks for taking a look. One other question that I have is whether or not there should be fewer Azure Pipelines. 

Other noarch feedstocks I've looked at only have `.azure-pipelines/azure-pipelines-linux.yml`. 

Should `.azure-pipelines/azure-pipelines-win.yml` and `.azure-pipelines/azure-pipelines-osx.yml` still be there? Should the rerendering delete them? Or should I be manually deleting them?

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipe/meta.yaml:15]: ah, i didn't see the `script` there.

i recommend not having any `script` in the top package, changing it to e.g. `cloudpathlib-split` and having the first output be `cloudpathlib`. You can then use the "extension" thing

```yaml
build: &build 
  number: 1 
  noarch: python

outputs:
  - name: cloudpathlib
    build:
      <<: *build
      script: {{ PYTHON }} -m pip install . -vv
```

Comment (bollwyvl) [reply recipe/meta.yaml:15]: You'd also then need: 

```yaml
extras:
  feedstock-name: cloudpathlib
```

at the end

**Comment by bollwyvl** [comment]
> One other question that I have is whether or not there should be fewer Azure Pipelines.

yes: once all your metadata are correct, it should just be the `linux-64` build

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

**Comment by jayqi** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (jayqi) [reply recipe/meta.yaml:15]: I keep getting YAML parser errors that aliases are undefined when trying to use the anchors/aliases. Seems to work if I just explicitly write out the duplicate information.

**Pull request merged**